### PR TITLE
DCOS_OSS 1783 Update modals to match UI Kit

### PR DESCRIPTION
**Pull request by leemunroe**
The goal of this PR is to align all modal designs with our UI Kit. In particular:

* Size of modals
* Layout (header, body, footer)
* Heading alignment and size
* Button alignment
* Wording used for headings and actions

This **does not address**:

* Adding a close icon button to each modal (separate feature request)
* Modals that could be full screen (e.g. Jobs)
* Modals that could have tabs instead of sidebars (e.g. LDAP directory)
* Form components or validation within the modals

In order to make this sweeping change, there are updates required in this repo, the private enterprise plugin repo and the reactjs-components repo.

### CODE REVIEW

**Comment by MatApple** [comment]
@leemunroe @ashenden I've moved the PR base from the feature branch to master so this doesn't  block the feature branch.

**Comment by leemunroe** [comment]
Alfred retest this please

**Comment by juliangieseke** [comment]
Alfred test this please

**Comment by leemunroe** [comment]
Alfred retest this please

**Comment by juliangieseke** [comment]
Alfred test this please

**Comment by bstavroulakis** [comment]
Multiple tests are failing due to the DOM class updates

**Comment by juliangieseke** [comment]
Hm, not sure if all fails are legit here, also there are some styling issues.

screen shot 2017-12-19 at 12 32 53
screen shot 2017-12-19 at 12 39 42

**Comment by leemunroe** [comment]
A bunch of tests are failing because it is looking for a class that is in the latest `reactjs-components` [master](https://github.com/mesosphere/reactjs-components/pull/430), but this hasn't been published to NPM yet. How do we deal with this scenario?

@juliangieseke Not sure what's happening with the header, need to look at that.

**Comment by orlandohohmeier** [comment]
@juliangieseke could you please help move this one forward?

**Comment by juliangieseke** [comment]
@leemunroe I have squashed your commits for easier handling and also resolved the conflicts. can you please pull this using `git pull --rebase` and run `npm install` from you `dcos-ui` root directory to make sure everything is correct?

**Comment by leemunroe** [comment]
Alfred retest this please

**Comment by leemunroe** [comment]
Alfred retest please

**Comment by leemunroe** [comment]
alfred test this please

**Comment by juliangieseke** [comment]
Its green 🤩

**Comment by leemunroe** [comment]
@juliangieseke Should be fixed with this reactjs-components PR https://github.com/mesosphere/reactjs-components/pull/433

**Comment by juliangieseke** [comment]
Alfred test this

File: src/js/components/modals/ModalHeading.js [hunk 1]
```diff
@@ -2,25 +2,19 @@ import classNames from "classnames";
 import React from "react";
 
 const ModalHeading = props => {
-  const { align, children, className, flush, level } = props;
+  const { children, level } = props;
```
Comment (DanielMSchmidt) [inline src/js/components/modals/ModalHeading.js]: We now have props in the propTypes and default props that are not used within the component, let's remove them 💪

Comment (juliangieseke) [reply src/js/components/modals/ModalHeading.js]: fixed 👍

**Comment by bstavroulakis** [comment]
Alfred test this

**Pull request merged**